fl4health/feature_alignment/tab_features_preprocessor.py

```diff
@@ -0,0 +1,147 @@
+from fl4health.feature_alignment.tabular_type import TabularType
+
+
+class TabularFeaturesPreprocessor:
+    """
+    TabularFeaturesPreprocessor is responsible for constructing
+    the appropriate column transformers based on the information
+    encoded in tab_feature_encoder. These transformers will
+    then be applied to a pandas dataframe.
+
+    Each tabular feature, which corresponds to a column
+    in the pandas dataframe, has its own column transformer. A default
+    transformer is initialized for each feature based on its data type,
+    but the user may also manually specify a transformer for this
+    feature.
+
+    Parameters
+    ----------
+    tab_feature_encoder: TabularFeaturesInfoEncoder
+        Encodes the information necessary for constructing the column transformers.
+    """
+
+    def __init__(self, tab_feature_encoder: TabularFeaturesInfoEncoder) -> None:
+        self.features_to_pipelines: Dict[str, Pipeline] = {}
+        self.targets_to_pipelines: Dict[str, Pipeline] = {}
+
+        self.tabular_features = tab_feature_encoder.get_tabular_features()
+        self.tabular_targets = tab_feature_encoder.get_tabular_targets()
+
+        self.feature_columns = tab_feature_encoder.get_feature_columns()
+        self.target_columns = tab_feature_encoder.get_target_columns()
+
+        self.initialize_default_pipelines(self.tabular_features, self.features_to_pipelines, one_hot=True)
+        self.initialize_default_pipelines(self.tabular_targets, self.targets_to_pipelines, one_hot=False)
+
+        self.data_column_transformer = self.return_column_transformer(self.features_to_pipelines)
+        self.target_column_transformer = self.return_column_transformer(self.targets_to_pipelines)
+
+    def get_default_numeric_pipeline(self) -> Pipeline:
+        return Pipeline(steps=[("imputer", SimpleImputer(strategy="mean")), ("scaler", MinMaxScaler())])
+
+    def get_default_binary_pipeline(self) -> Pipeline:
+        return Pipeline(steps=[("imputer", SimpleImputer(strategy="most_frequent")), ("encoder", OrdinalEncoder())])
+
+    def get_one_hot_pipeline(self, categories: MetaData) -> Pipeline:
+        return Pipeline(steps=[("encoder", OneHotEncoder(handle_unknown="ignore", categories=[categories]))])
+
+    def get_ordinal_pipeline(self, categories: MetaData) -> Pipeline:
+        return Pipeline(
+            steps=[
+                (
+                    "encoder",
+                    OrdinalEncoder(
+                        unknown_value=len(categories) + 1,
+                        handle_unknown="use_encoded_value",
+                        categories=[categories],
+                    ),
+                )
+            ]
+        )
+
+    def get_default_string_pipeline(self, vocabulary: MetaData) -> Pipeline:
+        return Pipeline(steps=[("vectorizer", TextColumnTransformer(TfidfVectorizer(vocabulary=vocabulary)))])
+
+    def initialize_default_pipelines(
```

Comment:
Since this function is an "initialization function, do we need the `feature_to_transformer` dict as an argument. I think we could just fill an empty on here and return it. Then the call would just initialize the pipelines as
```
self.features_to_pipelines: Dict[str, Pipeline] = self.initialize_default_pipelines(self.tabular_features, self.features_to_pipelines, one_hot=True)
self.targets_to_pipelines: Dict[str, Pipeline] = self.initialize_default_pipelines(self.tabular_targets, self.targets_to_pipelines, one_hot=False)
```
in the constructor.